awx/main/management/commands/cleanup_jobs.py

```diff
@@ -87,7 +87,8 @@ def cleanup_jobs(self):
 
     def cleanup_ad_hoc_commands(self):
         skipped, deleted = 0, 0
-        for ad_hoc_command in AdHocCommand.objects.all():
+        ad_hoc_commands = AdHocCommand.objects.filter(created__lte=self.cutoff)
+        for ad_hoc_command in ad_hoc_commands:
```

Comment:
I'd also suggest adding `.iterator()` to all of these querysets to minimize memory usage.